guidelines/backend-coding-standard/backend-coding-standard.md

```diff
@@ -0,0 +1,40 @@
+# Backend Coding Standard Draft
+
+## Coding style
+- function should be stateless, pure
+- function should not have side effect, i.e. interact with the environment outside the function, causing mutatation with other things that is not expected to change
+- input id instead of the object itself to the function, then query by id inside the function
+- function name should be describing the business action instead of implementation detail (except for private functions)
+  - e.g. prefer `consume_discount_code` to `decrease_discount_code_quota_by_one`
+- be caution of n + 1 query
+- consider usage of in memory or query from db
+  - e.g. `select` / `where`
```

Comment:
These two points are more like a reminder only without guiding developer what to do and how to follow.
Need to be more concise on what we suggest people doing here and provide example to support it.

Think of yourself as a reader of this document, it you don't know what to do after reading a point, then it mean the communication is failed.